### ISSUE TITLE
Installation to AWS EKS using helm fails

### ISSUE THREAD

**Issue by cardiff1982** [kubernetes/autoscaler#4762]
**Which component are you using?**:

cluster-autoscaler

**What version of the component are you using?**:
1.21

Component version:

**What k8s version are you using (`kubectl version`)?**:

<details><summary><code>kubectl version</code> Output</summary><br><pre>
$ kubectl version

</pre></details>

**What environment is this in?**:
AWS

**What did you expect to happen?**:

**What happened instead?**:

Got strange error, while trying to install autoscaler with helm chart. Installing to the same cluster with the same role, but just with applying .yaml manifests is OK. 
Executed command:
helm -n ops install autoscaler autoscaler/cluster-autoscaler --set rbac.serviceAccount.annotations."eks\.amazonaws\.com/role-arn"=<ROLE_ARN> --set autoDiscovery.clusterName=<CLUSTER_NAME> --set awsRegion=eu-west-1 --create-namespace

Autoscaler POD logs:

goroutine 249 [select]:
k8s.io/client-go/tools/cache.(*processorListener).pop(0xc0008bca80)
	/gopath/src/k8s.io/autoscaler/cluster-autoscaler/vendor/k8s.io/client-go/tools/cache/shared_informer.go:752 +0x156
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
	/gopath/src/k8s.io/autoscaler/cluster-autoscaler/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:73 +0x5a
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/gopath/src/k8s.io/autoscaler/cluster-autoscaler/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x88

goroutine 295 [sync.Cond.Wait]:
sync.runtime_notifyListWait(0xc0006f0e00, 0x0)
	/usr/local/go/src/runtime/sema.go:513 +0x13d
sync.(*Cond).Wait(0xc000496000)
	/usr/local/go/src/sync/cond.go:56 +0x8c
golang.org/x/net/http2.(*pipe).Read(0xc0006f0de8, {0xc0006ba200, 0x200, 0x200})
	/gopath/src/k8s.io/autoscaler/cluster-autoscaler/vendor/golang.org/x/net/http2/pipe.go:65 +0xeb
golang.org/x/net/http2.transportResponseBody.Read({0x1}, {0xc0006ba200, 0x0, 0xc00096fcb0})
	/gopath/src/k8s.io/autoscaler/cluster-autoscaler/vendor/golang.org/x/net/http2/transport.go:2104 +0x77
encoding/json.(*Decoder).refill(0xc0006cfe00)
	/usr/local/go/src/encoding/json/stream.go:165 +0x17f
encoding/json.(*Decoder).readValue(0xc0006cfe00)
	/usr/local/go/src/encoding/json/stream.go:140 +0xbb
encoding/json.(*Decoder).Decode(0xc0006cfe00, {0x350a880, 0xc0005f3c80})
	/usr/local/go/src/encoding/json/stream.go:63 +0x78
k8s.io/apimachinery/pkg/util/framer.(*jsonFrameReader).Read(0xc0006fd770, {0xc0006b9400, 0x400, 0x400})
	/gopath/src/k8s.io/autoscaler/cluster-autoscaler/vendor/k8s.io/apimachinery/pkg/util/framer/framer.go:152 +0x19c
k8s.io/apimachinery/pkg/runtime/serializer/streaming.(*decoder).Decode(0xc000092280, 0x0, {0x420a720, 0xc00039f2c0})
	/gopath/src/k8s.io/autoscaler/cluster-autoscaler/vendor/k8s.io/apimachinery/pkg/runtime/serializer/streaming/streaming.go:77 +0xa7
k8s.io/client-go/rest/watch.(*Decoder).Decode(0xc000359560)
	/gopath/src/k8s.io/autoscaler/cluster-autoscaler/vendor/k8s.io/client-go/rest/watch/decoder.go:49 +0x4f
k8s.io/apimachinery/pkg/watch.(*StreamWatcher).receive(0xc00039f280)
	/gopath/src/k8s.io/autoscaler/cluster-autoscaler/vendor/k8s.io/apimachinery/pkg/watch/streamwatcher.go:105 +0x11c
created by k8s.io/apimachinery/pkg/watch.NewStreamWatcher
	/gopath/src/k8s.io/autoscaler/cluster-autoscaler/vendor/k8s.io/apimachinery/pkg/watch/streamwatcher.go:76 +0x135

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a cluster in AWS EKS
2. Create IAM OIDC provider for this cluster
3. Create a role and policy, described in documentation
4. Try to install with helm

**Comment by meierj-wayscript** [comment]
Experiencing the same thing.

**Comment by michelzanini** [comment]
I experienced this problem too with the latest Helm chart version.
I was upgrading from `9.11.0` to `9.19.0` and got the error above as well.

It turned out that the AWS IAM permissions required have changed to require a little more.
In my case I had:

```
"autoscaling:DescribeAutoScalingGroups",
"autoscaling:DescribeAutoScalingInstances",
"autoscaling:DescribeLaunchConfigurations",
"autoscaling:DescribeTags",
"ec2:DescribeLaunchTemplateVersions",
"autoscaling:SetDesiredCapacity",
"autoscaling:TerminateInstanceInAutoScalingGroup",
"autoscaling:UpdateAutoScalingGroup"
```

And there was two more required permissions required:

```
"ec2:DescribeInstanceTypes",
"eks:DescribeNodegroup"
```

Once those were added everything worked fine.

PS: I simplified the permissions above just to list them. On the IAM policy they are split into statements etc.

**Comment by davidroth** [comment]
@michelzanini Thanks for the hint regarding the new permissions required. This fixed it for me after I upgraded to `9.19.3`.

It seems the documation on the AWS page is outdated: https://docs.aws.amazon.com/eks/latest/userguide/autoscaling.html#cluster-autoscaler

`"ec2:DescribeInstanceTypes", "eks:DescribeNodegroup" is not listed there.

**Comment by jonny-rimek** [comment]
I encocuntered a similar issue when upgrading from `9.19.4` upwards, two more permissions are needed. for full policy check [here](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/README.md#iam-policy)
